examples/tutorials/CartPole/CartPole_Tutorial.md

```diff
@@ -0,0 +1,537 @@
+
+```python
+    def __init__(self, config = None):
+        super().__init__(config)
+
+        self.x_threshold = 100
+        self.theta_threshold_radians = self.config["max_move"] * math.pi / 180
+        self.config.update({'max_angle': self.theta_threshold_radians})
+        
+        high = np.array(
+            [
+                self.x_threshold * 2,
+                np.finfo(np.float32).max,
+                self.theta_threshold_radians,
+                np.finfo(np.float32).max,
+            ],
+            dtype=np.float32,
+        )
+
+        nb_actions = 2
+        self.action_space = spaces.Discrete(nb_actions)
+        self.nb_actions = str(nb_actions)
+
+        self.observation_space = spaces.Box(high, -high, dtype=np.float32)
+```
+
+The second part of is to override the `step` and `reset` methods from the `AbstractEnv` class. For the step method, no additions are required. For the reset method, we need to restart the scene using `start_scene` and return the first observation. It might also be needed to update some parameters or configs like the randomized goal position in the case of some environments.
+
+```python
+    def step(self, action):
+        return super().step(action)
+
+    def reset(self):
+        """Reset simulation.
+
+        Note:
+        ----
+            We launch a client to create the scene. The scene of the program is
+            client_<scene>Env.py.
+
+        """
+        super().reset()
+
+        self.config.update({'goalPos': self.goal})
+
+        obs = start_scene(self.config, self.nb_actions)
+        
+        return np.array(obs['observation'])
+```
+
+
+# Step 2: Create the Environment's Toolbox
+
+The next step is to implement the methods and functions necessary for the RL algorithms to work. Three essential parts need to be implemented for this, defining and applying the actions, defining and calculating the reward, and getting the new observation from the simulation and updating it. One additional component that is relevant to some environments is a `GoalSetter` to define and update a target goal if necessary.
+
+First, we need to creat a new file `CartPoleToolbox`. For each of the parts that need to be implemented, some components are required to be defined to make the SOFA scene compatible with SofaGym.
+- Actions:
+  - startCmd function
+- Reward:
+  - getReward function
+  - RewardShaper class
+- Observation:
+  - getState function
+  - getPos function
+  - setPos function
+- Goal:
+  - GoalSetter class
+
+## Actions
+The possible actions that could be applied by the agent must be defined. The CartPole environment has two actions as discussed in the previous step, move the cart left or right. This could be achieved by applying a constant force field on the cart with a positive value (right direction) or a negative value (left direction) depednding on the chosen action 0 or 1, respectively.
```

Comment:
```suggestion
The possible actions that could be applied by the agent must be defined. The CartPole environment has two actions as discussed in the previous step, move the cart left or right. This could be achieved by applying a constant force field on the cart with a positive value (right direction) or a negative value (left direction) depending on the chosen action 0 or 1, respectively.
```
Could you run a spell checker on the whole file?

Reply:
done